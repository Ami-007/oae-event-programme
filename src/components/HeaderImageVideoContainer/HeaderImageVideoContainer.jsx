import YoutubeEmbed from "../YoutubeEmbed/YoutubeEmbed";
import "./HeaderImageVideoContainer.scss";

const HeaderImageVideoContainer = (props) => {
  const { content, isVideo } = props;

  // two props, content and isVideo
  // check to see if content is true - if not return null/console error
  // if true, return either yt embed or img

  if (!content) {
    console.error("No props provided to header image/video container.");
    return null;
  }

  return <div className="header-container">{isVideo ? <YoutubeEmbed embedId={content} /> : <img src={content} className="header-image" alt="Event image" />}</div>;
};

export default HeaderImageVideoContainer;
