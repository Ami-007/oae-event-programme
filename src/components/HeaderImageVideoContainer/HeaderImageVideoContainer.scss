@use "../../assets/sass/variables.scss" as *;

.header-image-video-container {
  background: linear-gradient(180deg, var(--theme-color-default) 50%, $color-white 50%);
  padding: 1rem;
  display: flex;
  justify-content: center;
}

.header-image {
  border-radius: 15px;
  max-width: 100%;
}
