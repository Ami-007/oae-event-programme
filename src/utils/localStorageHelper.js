export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
