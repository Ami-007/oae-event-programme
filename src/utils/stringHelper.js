export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
