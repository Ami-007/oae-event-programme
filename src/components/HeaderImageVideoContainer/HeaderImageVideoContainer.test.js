import { customRender } from "../../utils/testUtils";
import placeHolderImage from "../../assets/images/image33.png";
import HeaderImageVideoContainer from "./HeaderImageVideoContainer";

test("it should render the component", () => {
  const { container } = customRender(<HeaderImageVideoContainer content={placeHolderImage} isVideo={false} />);
  expect(container).toMatchSnapshot();
});

test("it should render the YoutubeEmbed component if isVideo is true", () => {
  const { container } = customRender(<HeaderImageVideoContainer content={placeHolderImage} isVideo={true} />);
  expect(container).toMatchSnapshot();
});

test("it should not render anything without props", () => {
  const { container } = customRender(<HeaderImageVideoContainer />);
  expect(container).toMatchSnapshot();
});
