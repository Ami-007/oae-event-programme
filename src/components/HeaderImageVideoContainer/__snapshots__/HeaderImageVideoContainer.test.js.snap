// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should not render anything without props 1`] = `<div />`;

exports[`it should render the component 1`] = `
<div>
  <div
    class="header-container"
  >
    <img
      alt="Event image"
      class="header-image"
      src="image33.png"
    />
  </div>
</div>
`;
